### PR TITLE
Support Mysql8.0

### PR DESCRIPTION
**Pull request by whydesd**

### CODE REVIEW

File: install/Mysql.sql [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  `url` varchar(200) default NULL,
+  `ip` varchar(64) default NULL,
+  `agent` varchar(200) default NULL,
+  `text` text,
+  `type` varchar(16) default 'comment',
+  `status` varchar(16) default 'approved',
+  `parent` int(10) unsigned default '0',
+  PRIMARY KEY  (`coid`),
+  KEY `cid` (`cid`),
+  KEY `created` (`created`)
```
Comment (whydesd) [inline install/Mysql.sql:37]: INN0B

File: install/Mysql.sql [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  `password` varchar(32) default NULL,
+  `commentsNum` int(10) unsigned default '0',
+  `allowComment` char(1) default '0',
+  `allowPing` char(1) default '0',
+  `allowFeed` char(1) default '0',
+  `parent` int(10) unsigned default '0',
+  PRIMARY KEY  (`cid`),
+  UNIQUE KEY `slug` (`slug`),
+  KEY `created` (`created`)
+) ENGINE=INNODB  DEFAULT CHARSET=%charset%;
```
Comment (whydesd) [inline install/Mysql.sql:67]: INN0B

File: install/Mysql.sql [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  `cid` int(10) unsigned NOT NULL,
+  `name` varchar(200) NOT NULL,
+  `type` varchar(8) default 'str',
+  `str_value` text,
+  `int_value` int(10) default '0',
+  `float_value` float default '0',
+  PRIMARY KEY  (`cid`,`name`),
+  KEY `int_value` (`int_value`),
+  KEY `float_value` (`float_value`)
+) ENGINE=INNODB  DEFAULT CHARSET=%charset%;
```
Comment (whydesd) [inline install/Mysql.sql:85]: INN0B

File: install/Mysql.sql [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  `name` varchar(200) default NULL,
+  `slug` varchar(200) default NULL,
+  `type` varchar(32) NOT NULL,
+  `description` varchar(200) default NULL,
+  `count` int(10) unsigned default '0',
+  `order` int(10) unsigned default '0',
+  `parent` int(10) unsigned default '0',
+  PRIMARY KEY  (`mid`),
+  KEY `slug` (`slug`)
+) ENGINE=INNODB  DEFAULT CHARSET=%charset%;
```
Comment (whydesd) [inline install/Mysql.sql:104]: INN0B

File: install/Mysql.sql [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+--
+-- 表的结构 `typecho_options`
+--
+
+CREATE TABLE `typecho_options` (
+  `name` varchar(32) NOT NULL,
+  `user` int(10) unsigned NOT NULL default '0',
+  `value` text,
+  PRIMARY KEY  (`name`,`user`)
+) ENGINE=INNODB DEFAULT CHARSET=%charset%;
```
Comment (whydesd) [inline install/Mysql.sql:117]: INN0B

File: install/Mysql.sql [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+--
+-- 表的结构 `typecho_relationships`
+--
+
+CREATE TABLE `typecho_relationships` (
+  `cid` int(10) unsigned NOT NULL,
+  `mid` int(10) unsigned NOT NULL,
+  PRIMARY KEY  (`cid`,`mid`)
+) ENGINE=INNODB DEFAULT CHARSET=%charset%;
```
Comment (whydesd) [inline install/Mysql.sql:129]: INN0B

File: install/Mysql.sql [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+  `screenName` varchar(32) default NULL,
+  `created` int(10) unsigned default '0',
+  `activated` int(10) unsigned default '0',
+  `logged` int(10) unsigned default '0',
+  `group` varchar(16) default 'visitor',
+  `authCode` varchar(64) default NULL,
+  PRIMARY KEY  (`uid`),
+  UNIQUE KEY `name` (`name`),
+  UNIQUE KEY `mail` (`mail`)
+) ENGINE=INNODB  DEFAULT CHARSET=%charset%;
```
Comment (whydesd) [inline install/Mysql.sql:152]: INN0B

**Comment by joyqi** [comment]
innodb引擎本来就是被支持的

**Pull request closed without merging**